conan/cache/cache.py

```diff
@@ -95,15 +82,6 @@ def create_tmp_package_layout(self, pref: ConanReference):
         self._create_path(package_path)
         return PackageLayout(pref, os.path.join(self.base_folder, package_path))
 
-    def create_reference_layout(self, ref: ConanReference):
```

Comment:
This was not called externally, so I have integrated it in the ``get_or_create_reference_layout()`` (it could have been made private, but as it is not very long, and only used once, I integrated it)